controllers/profile.js

```diff
@@ -1,61 +1,251 @@
+    try {
+        const { idDate } = req.params;
+        const user = await Doctor.findOne({
+            userData: req.user.userId,
+            workTime: idDate,
+        });
+        if (user) {
+            await Doctor.findOneAndUpdate(
+                {
+                    userData: req.user.userId,
+                },
+                {
+                    $pullAll: { workTime: [idDate] },
+                }
+            );
+            await DateTime.deleteOne({
+                _id: idDate,
+            });
+            res.status(200).json(idDate);
+        }
+    } catch (e) {
+        console.log(e);
+        res.send({ message: constants.ERRORS_MESSAGE.SERVER_ERROR });
+    }
+};
+
+exports.profilePatientAddAppointmentPut = async (req, res) => {
+    try {
+        const user = await Patient.findOne({
+            userData: req.user.userId,
+        });
+        const dateTime = await DateTime.findOne({
+            _id: req.body.data,
+        });
+
+        if (user.visit.indexOf(req.body.data) > -1) {
+            return res.status(400).json({
+                message: `You already have an appointment at this time.`,
+            });
+        }
+        if (user && dateTime) {
+            await Patient.findOneAndUpdate(
+                {
+                    userData: req.user.userId,
+                },
+                {
+                    $push: { visit: dateTime.id },
+                }
+            );
+
+            const date = await DateTime.findOne({
+                _id: req.body.data,
+            }).populate({
+                path: "doctorData",
+                select: ["direction", "workPlace", "userData"],
+
+                populate: [
+                    { path: "workPlace", select: ["name", "address"] },
+                    { path: "userData", select: ["name", "lastName"] },
+                ],
+            });
+            if (date) {
+                await DateTime.findOneAndUpdate(
+                    {
+                        _id: req.body.data,
+                    },
+                    {
+                        $set: {
+                            patientData: user.id,
+                        },
+                    }
+                );
+                const response = {
+                    doctorName: date.doctorData.userData.name,
+                    doctorLastName: date.doctorData.userData.lastName,
+                    doctorDirection: date.doctorData.direction,
+                    medCenterName: date.doctorData.workPlace.name,
+                    medCenterAddress: date.doctorData.workPlace.address,
+                    time: date.time,
+                    date: date.date,
+                };
+
+                res.json(response);
+            }
+        }
+    } catch (e) {
+        console.log(e);
+        res.send({ message: constants.ERRORS_MESSAGE.SERVER_ERROR });
+    }
+};
+exports.profilePatientDeleteAppointmentDelete = async (req, res) => {
```

Comment:
add lines between the controllers for better readability

Reply:
fixed